hypothesis-python/src/hypothesis/internal/conjecture/datatree.py

```diff
@@ -55,7 +61,7 @@ def inconsistent_generation():
     )
 
 
-EMPTY: frozenset = frozenset()
+EMPTY: frozenset[int] = frozenset()
```

Comment:
I think we should revisit this later rather than growing the present PR, but seeing `int` everywhere is starting to make me itch for some `NewType`s 😅

Reply:
NewType how? Just to enforce that all integers used in datatree.py were created in datatree.py?

Reply:
To distinguish integers used to index the choice sequence vs the datatree vs the possible-choices-at-this-position, etc.  I'm not confident it would be a net-improvement, I've never made much use of newtypes in python, but at some point we might want to try.  (we could always accept `:int | ChoiceIndex` as an argument while returning `-> ChoiceIndex` from the relevant functions, to keep it as ~documentation only)

Reply:
gotcha, that could be a nice improvement. I'll play around with it once things are more completely typed